compiler/optimizer/ValuePropagationCommon.cpp

```diff
@@ -2606,7 +2607,7 @@ void OMR::ValuePropagation::generateArrayTranslateNode(TR::TreeTop *callTree,TR:
 
       termCharNode = TR::Node::create(callNode,TR::iconst, 0, termchar);
       }
-   else if (isSBCSEncoder) //only z
+   else if (isSBCSEncoder || isEncodeFromLatin1) //only z
       {
       arrayTranslateNode->setTermCharNodeIsHint(false);
```

Comment:
Is this one supposed to stop when it hits a zero? (unlike `decodeToLatin1Impl()` from #7758)